Core/Tornado/Server/TornadoService.py

```diff
@@ -0,0 +1,538 @@
+      We have several try except to catch the different problem who can occurs
+
+      - First, the method does not exist => Attribute error, return an error to client
+      - second, anything happend during execution => General Exception, send error to client
+    """
+
+    # getting method
+    try:
+      method = getattr(self, 'export_%s' % self.method)
+    except AttributeError as e:
+      self._httpError = HTTPErrorCodes.HTTP_NOT_IMPLEMENTED
+      return S_ERROR("Unknown method %s" % self.method)
+
+    # Decode args
+    args_encoded = self.get_body_argument('args', default=encode([]))
+
+    args = decode(args_encoded)[0]
+    # Execute
+    try:
+      self.initializeRequest()
+      retVal = method(*args)
+    except Exception as e:  # pylint: disable=broad-except
+      gLogger.exception("Exception serving request", "%s:%s" % (str(e), repr(e)))
+      retVal = S_ERROR(repr(e))
+      self._httpError = HTTPErrorCodes.HTTP_INTERNAL_SERVER_ERROR
+
+    return retVal
+
+  def __write_return(self, dictionary):
+    """
+      Write to client what we wan't to return to client
+      It must be a dictionary
+    """
+
+    # In case of error in server side we hide server CallStack to client
+    if 'CallStack' in dictionary:
+      del dictionary['CallStack']
+
+    # Write status code before writing, by default error code is "200 OK"
+    self.set_status(self._httpError)
+
+    # This is basically only used for file download through
+    # the 'streamToClient' method.
+    if self.rawContent:
+      # See 4.5.1 http://www.rfc-editor.org/rfc/rfc2046.txt
+      self.set_header("Content-Type", "application/octet-stream")
+      returnedData = dictionary
+    else:
+      self.set_header("Content-Type", "application/json")
+      returnedData = encode(dictionary)
+
+    self.write(returnedData)
+
+  def reportUnauthorizedAccess(self, errorCode=401):
+    """
+      This method stop the current request and return an error to client
+
+
+      :param int errorCode: Error code, 403 is "Forbidden" and 401 is "Unauthorized"
+    """
+    error = S_ERROR(ENOAUTH, "Unauthorized query")
+    gLogger.error(
+        "Unauthorized access to %s: %s from %s" %
+        (self.request.path,
+         self.credDict['DN'],
+         self.request.remote_ip))
+
+    self._httpError = errorCode
+    self.__write_return(error)
+    self.finish()
+
+  def on_finish(self):
+    """
+      Called after the end of HTTP request
+    """
+    requestDuration = time.time() - self.requestStartTime
+    gLogger.notice("Ending request to %s after %fs" % (self.srv_getURL(), requestDuration))
+
+  def _gatherPeerCredentials(self):
+    """
+      Load client certchain in DIRAC and extract informations.
+
+      The dictionary returned is designed to work with the AuthManager,
+      already written for DISET and re-used for HTTPS.
+    """
+
+    # This line get certificates, it must be change when M2Crypto will be fully integrated in tornado
```

Comment:
So what now?

Reply:
I removed the comment , I already changed it to what it should be